integration-libs/cdc/manage-users/cdc-user-list.service.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { CdcJsService } from '@spartacus/cdc/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  WindowRef,
+} from '@spartacus/core';
+import { B2BUserService } from '@spartacus/organization/administration/core';
+import { TableService } from '@spartacus/storefront';
+import { UserListService } from '@spartacus/organization/administration/components';
+import { Subscription } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CdcUserListService extends UserListService {
+  constructor(
+    protected tableService: TableService,
+    protected userService: B2BUserService,
+    protected globalMessageService: GlobalMessageService,
+    protected winRef: WindowRef,
+    protected cdcJsService: CdcJsService
+  ) {
+    super(tableService, userService);
+  }
+  protected subscription: Subscription = new Subscription();
+
+  onCreateButtonClick(): void {
+    this.cdcJsService.didLoad().subscribe((cdcLoaded) => {
+      if (cdcLoaded) {
+        // get current organization ID using CDC Gigya SDK
+        this.cdcJsService.getOrganizationContext().subscribe({
+          next: (response) => {
```

Comment:
Instaed of doing inner subscriptions, which are harder to read, lets simplify this code, by using: `pipe` and `switchMap`
```ts
  const sub = this.cdcJsService.didLoad().pipe(
    switchMap(cdcLoaded => {
       /* ... */
       return this.cdcJsService.getOrganizationContext()
    }),
    tap(response => {
      /*...*/
       this.cdcJsService.openDelegatedAdminLogin(response.orgId);
    })
  ).subscribe(); // we subscribe only once here

  // moreover, let's make sure to cleanup subscriptions in case when an user leaves (destroys) this Component but the observable logic would not ever complete:
  this.subscription.add(sub);


/* ... */
// and eventually on component's destroy let's make sure do dispose any possible pending subscriptions
ngOnDestroy(){
  this.subscription?.unsubscribe();
}  
```

Reply:
done